### PR TITLE
[DevOps] Create the Bot Deployment Task

### PR DESCRIPTION
**Pull request by ceciliaavila**
## Description
Added a new task to deploy a Bot to Azure and connect it with channels. 

The task executes the '_az deployment create_' command to create the Resource Group and within it, the App Service, the App Service Plan, and the Bot Channels Registration. 

It also deploys a zipped bot and connects it to _Direct Line_, _Slack_, _MS Teams_, _Webex_, _Twilio_, and _Facebook_ channels.

image

## Testing
The following image shows the task running successfully:
image

And the next image shows the task failing when one of the parameters is invalid:
image

_Note: The [TFX extension](https://marketplace.visualstudio.com/items?itemName=ms-devlabs.vsts-developer-tools-build-tasks&targetId=89aec069-f48a-4934-b6d0-ac8472627737) needs to be installed to be able to run the publishing steps added in the yaml files._

### CODE REVIEW

**Comment by BruceHaley** [comment]
Looks good. I like it!
As you mentioned in your email, it requires the [TFX Extension](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmarketplace.visualstudio.com%2Fitems%3FitemName%3Dms-devlabs.vsts-developer-tools-build-tasks%26targetId%3D89aec069-f48a-4934-b6d0-ac8472627737&data=02%7C01%7Cv-bruhal%40microsoft.com%7C74676db5482e458fa31608d79deb47c3%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637151506170606964&sdata=vlIeOUs0UsrzdLqf3MMiC52kO4ifWANlifCuMrVrwY4%3D&reserved=0) dependency from Marketplace as a prerequisite.

File: build/botframework-cli-mac.yml [hunk 1]
```diff
@@ -79,6 +84,47 @@ jobs:
       inputs:
         artifactName: 'drop'
 
+    - powershell: |
```
Comment (munozemilio) [inline build/botframework-cli-mac.yml]: Can we add the task build steps in a different file please?

Comment (ceciliaavila) [reply build/botframework-cli-mac.yml]: Feedback applied. Thanks!

**Comment by cleemullins** [comment]
This is a nice PR that needs more attention. Converting to draft right now until we have time to look at it.

**Comment by tracyboehrer** [comment]
@ceciliaavila Is this stil a thing?

**Comment by ceciliaavila** [comment]
> @ceciliaavila Is this stil a thing?

I think we can close it. This was a DevOps task to ease the deployment of bots, but it's outdated now.

**Pull request closed without merging**